### ISSUE TITLE
HiDPI / Font support

### ISSUE THREAD

**Issue by smartattack** [OpenXenManager/openxenmanager#106]
Hi,

  I am using a few systems with HiDPI and the UI renders oddly.  Is there a way to fix this with settings, or would it be possible to add settings to allow changing the font sizes or scaling?

![hidpi_rendering](https://cloud.githubusercontent.com/assets/8866501/12148086/7a5237fa-b46a-11e5-9b2e-6c824d22c838.png)

**Comment by dlintott** [comment]
I think this is something that should get fixed when we eventually transition to GTK3 (not a small task!)

https://www.reddit.com/r/linux/comments/1n74un/will_only_gtk3_apps_receive_support_for_hidpi/

**Comment by smartattack** [comment]
Okay, understood.  Is there any way to influence the font rendering as-is with the current app?

**Comment by dlintott** [comment]
@smartattack I'm not sure on that one... I shall have to look into it

**Comment by dirk103** [comment]
I'm going to fix up the .glade files. Everything is a fixed sized and isn't designed to automatically expand.
Shouldn't take too long.
I'll do my best to make sure I don't break anything. Any tips for a beginner to git/open source contributing?
Do I have to create a separate branch for working on this?

Also, glade-gtk2 that I'm using crashes every five seconds, I'm going to go look for the most recent source to this, any advise?

**Comment by dlintott** [comment]
@dirk103 Many thanks for offering your help with the project.

As a starting point, you might want to check out the following article https://help.github.com/articles/fork-a-repo/

Forking the repo allows you to work on it, make changes, without affecting this one until you're happy with your changes when you can send a pull request.

No idea on why Glade is crashing, but it's reasonably simple to build from source which you can find here http://lists.ximian.com/pipermail/glade-devel/2014-May/002063.html

We have to use the older 3.8 version, as you've probobably discovered the project hasn't moved to GTK-3 yet. 

Hope that helps

**Comment by smartattack** [comment]
Yes, thank you for offering to look at this.  If I knew where to look I
would be happy to take a stab at it myself.  Are you going to update the
glade files to auto-scale or change things in some other way?

**Comment by dirk103** [comment]
@dlintott Yeah man. Thanks for a link. I'll fork it today. I have built Glade 3.8.5 from source. It seems to be working fine now, when I first started using it it was crashing and glitching all the time.

@smartattack I'm going to restructure the widgets in the glade files. Most of the layout uses fixed values which is very inflexible and relies on you having the exact font the glade files were designed with.
For now I'm just going to get everything in a usable state. I've discovered a couple things that will need to be reworked but thankfully it isn't too bad.

:)

**Comment by dlintott** [comment]
@dirk103 glad to hear that's working now. There are some oddities in the way the UI was originally structured which I started to rework as well as splitting the glade files up... Originally everything was in one file!

**Comment by smartattack** [comment]
I did manage to figure something out tonight:

If I create ~/.gtkrc-2.0  and populate it with this, it helps a lot.  The
fonts are smaller so they're tiny, but at least they fit within the spaces
meant for them so they aren't clipped:

style "openxenmanager"
{
font_name = "7"
}
widget_class "*" style "openxenmanager"
gtk-font-name = "7"

**Comment by dirk103** [comment]
@smartattack I hate hacks.

Download my fixed repo. I haven't went over this hugely. It's a real work in progress.

https://github.com/dirk103/openxenmanager

I have only touched a few dialogs yet, I think some of the main ones (that I use).

!! Please let me know if there are any specific windows you would like me to fix immediately !!

I'm in the process of learning XenServer right now so I'm doing that as well as fixing this client up!

**Comment by smartattack** [comment]
Wow, nice work.  It looks much better.  I can still see a few dialogs like
adding a network interface that have some issues but the main windows look
fine now.  I really appreciate your effort, thanks!

**Comment by dirk103** [comment]
Thanks man :)

Yeah, I'm working my way through the whole thing. I'd appreciate it if you could tell me the dialogs to prioritize. I'm just learning about Citrix Xen Server right now so I'm not completely sure what's all important.

I haven't dug into the code besides making sure I don't break things, but does this project use a base library to manipulate the Xen server in a standard protocol, or are you implementing that as well?

**Comment by smartattack** [comment]
I just had another look at your repo today, James.  The gui looks much more
polished and the font sizing is much better on my hidpi laptop.  Really
nice work, thanks!

**Comment by dirk103** [comment]
Thank you :)

I need to fix a few things up to get it accepted to the main repo. I'm hoping to redo the graphing, but need to learn some more Python and GTK first. 

Can I ask what system you're running it on?

**Comment by smartattack** [comment]
I am running linux Mint Cinnamon 17.3.  My display resolution is 2560x1440,
fwiw.

**Comment by jsarasin** [comment]
Probably close this issue hey?

**Comment by davidak** [comment]
@jsarasin is it fixed?

**Comment by jsarasin** [comment]
@davidak I believe all of the dialogs have been fixed up. @smartattack seemed to be happy. Of course there may be things here and there that could be fixed up, but more specific issues would have to be raised. Do you know of a place where scaling is an issue?

**Comment by davidak** [comment]
@jsarasin i don't work at the company anymore where we used Xen.
